File: hayagriva/dev-scratch/apply_branding.js
const fs = require('fs');
const path = require('path');

const repoRoot = '/Users/atulgrover/Desktop/HAYAGRIVA';
const resourcesDir = path.join(repoRoot, 'resources/resources');

const whiteLogoPng = path.join(resourcesDir, 'logo_white.png');
const whiteIconPng = path.join(resourcesDir, 'icon_white_512.png');
const base64White = fs.readFileSync(path.join(resourcesDir, 'logo_white_base64.txt'), 'utf8');

function main() {
    console.log('Applying all-white transparent Hayagriva branding...');

    // 1. Update resources/resources/preload.html with clean white pulsing animation
    const preloadPath = path.join(resourcesDir, 'preload.html');
    let preloadContent = fs.readFileSync(preloadPath, 'utf8');
    
    // Replace SVG element with IMG tag using base64 src
    const svgRegex = /<svg id="spinner"[\s\S]*?<\/svg>/;
    const newImgTag = `<img id="spinner" src="data:image/png;base64,${base64White}" alt="Hayagriva Logo" />`;
    preloadContent = preloadContent.replace(svgRegex, newImgTag);

    // Replace the SVG CSS styling with IMG styling
    preloadContent = preloadContent.replace(
        /\.custom-spinner\s+img\s*\{[\s\S]*?\}/,
        `.custom-spinner img {
            width: 32vw;
            height: auto;
            max-width: 450px;
            animation-delay: 0;
            animation-duration: 3s;
            animation-iteration-count: infinite;
            animation-name: theia-ide-spinner;
            animation-timing-function: ease-in-out;
        }`
    );

    // Modify the animation to have a pulsing scale and fading opacity in pure white (no color-shift filters)
    preloadContent = preloadContent.replace(
        /@keyframes\s+theia-ide-spinner\s*\{[\s\S]*?\}/g,
        `@keyframes theia-ide-spinner {
            0% {
                transform: scale(1.0);
                opacity: 0.95;
                filter: drop-shadow(0 0 10px rgba(255, 255, 255, 0.15));
            }
            50% {
                transform: scale(0.96);
                opacity: 0.55;
                filter: drop-shadow(0 0 16px rgba(255, 255, 255, 0.3));
            }
            100% {
                transform: scale(1.0);
                opacity: 0.95;
                filter: drop-shadow(0 0 10px rgba(255, 255, 255, 0.15));
            }
        }`
    );

    fs.writeFileSync(preloadPath, preloadContent, 'utf8');
    console.log(`- Updated preload.html: ${preloadPath}`);

    // 2. Generate new resources/resources/TheiaIDESplash.svg with transparent background and solid white mask
    const splashSvgPath = path.join(resourcesDir, 'TheiaIDESplash.svg');
    const splashSvgContent = `<?xml version="1.1" encoding="UTF-8" standalone="no"?>
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 1160 484" width="445.5" height="186">
  <defs>
    <!-- Define a mask using the base64 white logo -->
    <mask id="logo-mask">
      <image href="data:image/png;base64,${base64White}" x="0" y="0" width="1160" height="484" />
    </mask>
  </defs>
  
  <!-- Rect filled with pure white (#ffffff), masked by the logo -->
  <rect x="0" y="0" width="1160" height="484" fill="#ffffff" mask="url(#logo-mask)" />
</svg>`;
    fs.writeFileSync(splashSvgPath, splashSvgContent, 'utf8');
    console.log(`- Generated transparent solid-white splash screen: ${splashSvgPath}`);

    // 3. Copy files to ide/theia-extensions/product
    const productIconsDir = path.join(repoRoot, 'ide/theia-extensions/product/src/browser/icons');
    if (fs.existsSync(productIconsDir)) {
        // Copy product logos
        fs.copyFileSync(whiteLogoPng, path.join(productIconsDir, 'TheiaIDE.png'));
        fs.copyFileSync(whiteLogoPng, path.join(productIconsDir, 'TheiaIDE-next.png'));
        console.log(`- Copied product logos to ${productIconsDir}`);

        // Copy product icons
        fs.copyFileSync(whiteIconPng, path.join(productIconsDir, '512-512.png'));
        fs.copyFileSync(whiteIconPng, path.join(productIconsDir, '512-512-next.png'));
        console.log(`- Copied product icons to ${productIconsDir}`);
    }

    // 4. Copy files to resources/resources/icons
    fs.copyFileSync(whiteIconPng, path.join(resourcesDir, 'icons/512x512.png'));
    fs.copyFileSync(whiteIconPng, path.join(resourcesDir, 'icons/WindowIcon/512-512.png'));
    console.log(`- Copied window and launcher icons to resources/resources/icons`);

    // 5. Copy to all app resources directories in ide/applications
    const apps = ['electron', 'electron-next', 'browser'];
    apps.forEach(app => {
        const appResDir = path.join(repoRoot, `ide/applications/${app}/resources`);
        if (fs.existsSync(appResDir)) {
            // Copy preload.html
            fs.copyFileSync(preloadPath, path.join(appResDir, 'preload.html'));
            // Copy TheiaIDESplash.svg
            fs.copyFileSync(splashSvgPath, path.join(appResDir, 'TheiaIDESplash.svg'));
            
            // Safe copy icons
            const icon512 = path.join(appResDir, 'icons/512x512.png');
            if (fs.existsSync(path.dirname(icon512))) {
                fs.copyFileSync(whiteIconPng, icon512);
            }
            const windowIconPath = path.join(appResDir, 'icons/WindowIcon/512-512.png');
            if (fs.existsSync(path.dirname(windowIconPath))) {
                fs.copyFileSync(whiteIconPng, windowIconPath);
            }
            console.log(`- Copied branding resources to ${appResDir}`);
        }
    });

    console.log('Branding apply completed successfully.');
}

main();

File: hayagriva/scripts/apply_branding.py
import os
import base64
import re

repo_root = "/Users/atulgrover/Desktop/HAYAGRIVA"

# Image paths
small_logo_path = os.path.join(repo_root, "resources", "resources", "hayagriva_small.png")
medium_logo_path = os.path.join(repo_root, "resources", "resources", "hayagriva_medium.png")
large_logo_path = os.path.join(repo_root, "resources", "resources", "hayagriva_large.png")

# Icon target paths
icon_targets = [
    os.path.join(repo_root, "ide", "applications", "electron", "resources", "icons", "WindowIcon", "512-512.png"),
    os.path.join(repo_root, "ide", "applications", "electron", "resources", "icons", "MacLauncherIcons", "icon.icon", "Assets", "icon.png"),
    os.path.join(repo_root, "resources", "resources", "icons", "WindowIcon", "512-512.png"),
    os.path.join(repo_root, "resources", "resources", "icons", "512x512.png"),
    os.path.join(repo_root, "resources", "resources", "icon_black_512.png"),
    os.path.join(repo_root, "resources", "resources", "icon_white_512.png"),
    os.path.join(repo_root, "resources", "resources", "logo_black.png"),
    os.path.join(repo_root, "resources", "resources", "logo_white.png"),
]

# Splash screen targets
splash_targets = [
    os.path.join(repo_root, "ide", "applications", "electron", "resources", "TheiaIDESplash.svg"),
    os.path.join(repo_root, "ide", "applications", "browser", "resources", "TheiaIDESplash.svg"),
    os.path.join(repo_root, "resources", "resources", "TheiaIDESplash.svg"),
]

def load_base64(file_path):
    with open(file_path, "rb") as image_file:
        return base64.b64encode(image_file.read()).decode('utf-8')

def apply_branding():
    print("[Branding Compiler] Starting application of Hayagriva horse logo...")
    
    # Verify files exist
    for p in [small_logo_path, medium_logo_path, large_logo_path]:
        if not os.path.exists(p):
            print(f"Error: Required file {p} does not exist.")
            return

    # Load base64 payloads
    print("  -> Loading and encoding logo assets...")
    small_b64 = load_base64(small_logo_path)
    medium_b64 = load_base64(medium_logo_path)
    large_b64 = load_base64(large_logo_path)

    # 1. Overwrite launcher PNG icons
    print("  -> Copying large horse logo to launcher target icons...")
    with open(large_logo_path, "rb") as src_f:
        large_bytes = src_f.read()
    for target in icon_targets:
        os.makedirs(os.path.dirname(target), exist_ok=True)
        with open(target, "wb") as dest_f:
            dest_f.write(large_bytes)
        print(f"     ✓ Wrote: {os.path.relpath(target, repo_root)}")

    # 2. Patch Splash Screen SVGs
    print("  -> Patching Splash Screen SVG files with base64 horse logo...")
    for splash_path in splash_targets:
        if os.path.exists(splash_path):
            with open(splash_path, "r", encoding="utf-8") as f:
                content = f.read()
            
            # Find the href attribute in the mask image
            # Pattern matches: href="data:image/png;base64,..."
            pattern = r'href="data:image/png;base64,[^"]+"'
            new_href = f'href="data:image/png;base64,{large_b64}"'
            updated_content, count = re.subn(pattern, new_href, content)
            
            if count > 0:
                with open(splash_path, "w", encoding="utf-8") as f:
                    f.write(updated_content)
                print(f"     ✓ Patched: {os.path.relpath(splash_path, repo_root)}")
            else:
                print(f"     ⚠️ Warning: No logo-mask href pattern matched in {splash_path}")
        else:
            print(f"     ⚠️ Warning: Splash screen not found at {splash_path}")

    # 3. Patch Settings Dashboard HTML
    print("  -> Patching settings-dashboard.html header with inline horse logo...")
    dashboard_path = os.path.join(repo_root, "hayagriva", "lib", "assets", "settings-dashboard.html")
    if os.path.exists(dashboard_path):
        with open(dashboard_path, "r", encoding="utf-8") as f:
            content = f.read()
        
        # Replace the header line
        target_header = """    <h2>
      <span>Hayagriva Case Settings</span>
      <span class="version-badge" id="vaultVer">Vault Status</span>
    </h2>"""
        replacement_header = f"""    <h2>
      <div style="display: flex; align-items: center; gap: 10px;">
        <img src="data:image/png;base64,{small_b64}" style="width: 24px; height: 24px; object-fit: contain;" />
        <span>Hayagriva Case Settings</span>
      </div>
      <span class="version-badge" id="vaultVer">Vault Status</span>
    </h2>"""
        
        # Check if already patched
        if "data:image/png;base64" in content:
            # Re-patching: replace the dynamic header structure
            content = re.sub(r'<h2>\s*<div.*?>.*?<span>Hayagriva Case Settings</span>.*?</div>\s*<span class="version-badge" id="vaultVer">Vault Status</span>\s*</h2>', replacement_header, content, flags=re.DOTALL)
            print("     ✓ Re-patched existing header in settings-dashboard.html")
        else:
            content = content.replace(target_header, replacement_header)
            print("     ✓ Patched new header in settings-dashboard.html")
            
        with open(dashboard_path, "w", encoding="utf-8") as f:
            f.write(content)

    # 4. Patch extension.ts in browser extension
    print("  -> Injecting logo CSS overrides into extension.ts...")
    extension_path = os.path.join(repo_root, "ide", "theia-extensions", "hayagriva", "src", "browser", "extension.ts")
    if os.path.exists(extension_path):
        with open(extension_path, "r", encoding="utf-8") as f:
            content = f.read()
        
        # We will insert constants and inject them into the style block
        # Let's locate the style element setup in onStart
        style_block_target = """    style.id = 'hayagriva-hide-filetype-icons';
    style.textContent = `
      .theia-FileStatNode:not(.theia-DirNode) .file-icon,
      .theia-FileStatNode:not(.theia-DirNode) .theia-FileStatIcon,
      .theia-FileStatNode:not(.theia-DirNode) [class*="file-icon"] {
          display: none !important;
      }
      #theia-open-editors-widget,
      .theia-open-editors-widget,
      .theia-NavigatorWidget > .p-Panel > .theia-open-editors-widget {
          display: none !important;
      }
    `;"""

        # Our custom CSS rules override the robot welcome icon and chatbot response icon
        replacement_style_block = f"""    style.id = 'hayagriva-hide-filetype-icons';
    style.textContent = `
      .theia-FileStatNode:not(.theia-DirNode) .file-icon,
      .theia-FileStatNode:not(.theia-DirNode) .theia-FileStatIcon,
      .theia-FileStatNode:not(.theia-DirNode) [class*="file-icon"] {{
          display: none !important;
      }}
      #theia-open-editors-widget,
      .theia-open-editors-widget,
      .theia-NavigatorWidget > .p-Panel > .theia-open-editors-widget {{
          display: none !important;
      }}
      
      /* --- HAYAGRIVA Custom Branding Logo Overrides --- */
      svg.theia-WelcomeMessage-Logo * {{
          display: none !important;
      }}
      svg.theia-WelcomeMessage-Logo {{
          background-image: url('data:image/png;base64,{medium_b64}') !important;
          background-size: contain !important;
          background-repeat: no-repeat !important;
          background-position: center !important;
          width: 120px !important;
          height: 120px !important;
      }}
      svg.theia-WelcomeMessage-Logo[width="64"] {{
          width: 64px !important;
          height: 64px !important;
      }}
      .theia-AgentAvatar.codicon-copilot::before {{
          content: "" !important;
      }}
      .theia-AgentAvatar.codicon-copilot {{
          background-image: url('data:image/png;base64,{small_b64}') !important;
          background-size: contain !important;
          background-repeat: no-repeat !important;
          background-position: center !important;
          width: 20px !important;
          height: 20px !important;
          display: inline-block !important;
      }}
    `;"""

        if style_block_target in content:
            content = content.replace(style_block_target, replacement_style_block)
            print("     ✓ Successfully patched styling overrides in extension.ts")
        elif "HAYAGRIVA Custom Branding Logo Overrides" in content:
            # Already patched, let's update it by regex to make sure it's clean
            # We can find the custom overrides block and update the base64 values
            pattern = r'/\* --- HAYAGRIVA Custom Branding Logo Overrides --- \*/.*?\.theia-AgentAvatar\.codicon-copilot .*?\}'
            content = re.sub(pattern, f"""/* --- HAYAGRIVA Custom Branding Logo Overrides --- */
      svg.theia-WelcomeMessage-Logo * {{
          display: none !important;
      }}
      svg.theia-WelcomeMessage-Logo {{
          background-image: url('data:image/png;base64,{medium_b64}') !important;
          background-size: contain !important;
          background-repeat: no-repeat !important;
          background-position: center !important;
          width: 120px !important;
          height: 120px !important;
      }}
      svg.theia-WelcomeMessage-Logo[width="64"] {{
          width: 64px !important;
          height: 64px !important;
      }}
      .theia-AgentAvatar.codicon-copilot::before {{
          content: "" !important;
      }}
      .theia-AgentAvatar.codicon-copilot {{
          background-image: url('data:image/png;base64,{small_b64}') !important;
          background-size: contain !important;
          background-repeat: no-repeat !important;
          background-position: center !important;
          width: 20px !important;
          height: 20px !important;
          display: inline-block !important;
      }}""", content, flags=re.DOTALL)
            print("     ✓ Updated existing styling overrides in extension.ts")
        else:
            print("     ⚠️ Warning: Could not locate target style element setup in extension.ts")

        with open(extension_path, "w", encoding="utf-8") as f:
            f.write(content)

    print("[Branding Compiler] Finished branding configuration successfully!\n")

if __name__ == "__main__":
    apply_branding()
